bentoml/service.py

```diff
@@ -223,29 +241,38 @@ def api_decorator(handler_cls, *args, **kwargs):
 
     from bentoml.handlers.base_handlers import BentoHandler
 
-    if not (inspect.isclass(handler_cls) and issubclass(handler_cls, BentoHandler)):
-        raise InvalidArgument(
-            "BentoService @api decorator first parameter must "
-            "be class derived from bentoml.handlers.BentoHandler"
+    def decorator(func):
+        _api_name = func.__name__ if api_name is None else api_name
+        _api_doc = (
+            (func.__doc__ or DEFAULT_API_DOC).strip() if api_doc is None else api_doc
         )
 
-    def decorator(func):
-        api_name = kwargs.pop("api_name", func.__name__)
-        api_doc = kwargs.pop("api_doc", func.__doc__ or DEFAULT_API_DOC).strip()
+        if input is None:
+            # support bentoml<=0.7
+            if not args or not (
+                inspect.isclass(args[0]) and issubclass(args[0], BentoHandler)
+            ):
+                raise InvalidArgument(
+                    "BentoService @api decorator first parameter must "
```

Comment:
“Specify required parameter `input` for BentoML service API or, with the legacy interface, use class derived from bentoml.handlers.BentoHandler as the first positional parameter”